### PR TITLE
Move resource updates patches test to calculator

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go [hunk 1]
```diff
@@ -53,289 +39,137 @@ func (fpp *fakePodPreProcessor) Process(pod apiv1.Pod) (apiv1.Pod, error) {
...
 
-func addResourceRequestPatch(index int, res, amount string) resource_admission.PatchRecord {
-	return resource_admission.PatchRecord{
-		"add",
-		fmt.Sprintf("/spec/containers/%d/resources/requests/%s", index, res),
-		resource.MustParse(amount),
-	}
+type fakePatchCalculator struct {
+	patches []resource_admission.PatchRecord
+	err     error
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: Done

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go [hunk 2]
```diff
@@ -53,289 +39,137 @@ func (fpp *fakePodPreProcessor) Process(pod apiv1.Pod) (apiv1.Pod, error) {
...
 }
 
-func addResourceLimitPatch(index int, res, amount string) resource_admission.PatchRecord {
-	return resource_admission.PatchRecord{
-		"add",
-		fmt.Sprintf("/spec/containers/%d/resources/limits/%s", index, res),
-		resource.MustParse(amount),
-	}
+func (c fakePatchCalculator) CalculatePatches(pod *apiv1.Pod, _ *vpa_types.VerticalPodAutoscaler) (
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: Done

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go [hunk 3]
```diff
@@ -53,289 +39,137 @@ func (fpp *fakePodPreProcessor) Process(pod apiv1.Pod) (apiv1.Pod, error) {
...
 			podJson:              []byte("{"),
 			namespace:            "default",
+			vpa:                  testVpa,
 			podPreProcessorError: nil,
-			recommendResources:   []vpa_api_util.ContainerResources{},
-			recommendAnnotations: vpa_api_util.ContainerToAnnotationsMap{},
 			expectError:          fmt.Errorf("unexpected end of JSON input"),
-		},
-		{
+		}, {
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: Ok, I changed it but I don't think that affects the readability at all, so maybe doesn't matter?

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go [hunk 4]
```diff
@@ -53,289 +39,137 @@ func (fpp *fakePodPreProcessor) Process(pod apiv1.Pod) (apiv1.Pod, error) {
...
 	}
 	for _, tc := range tests {
 		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
 			fppp := fakePodPreProcessor{e: tc.podPreProcessorError}
-			fvm := fakeVpaMatcher{}
-			frp := fakeRecommendationProvider{tc.recommendResources, tc.recommendAnnotations, tc.recommendError}
-			cs := []patch.Calculator{patch.NewResourceUpdatesCalculator(&frp), patch.NewObservedContainersCalculator()}
-			h := NewResourceHandler(&fppp, &fvm, cs)
+			fvm := fakeVpaMatcher{vpa: tc.vpa}
+			h := NewResourceHandler(&fppp, &fvm, tc.calculators)
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: AssertEqPatch does not print the index of patch.

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+}
+
+func TestClalculatePatches_ResourceUpdates(t *testing.T) {
+	tests := []struct {
+		name                 string
+		pod                  *core.Pod
+		namespace            string
+		recommendResources   []vpa_api_util.ContainerResources
+		recommendAnnotations vpa_api_util.ContainerToAnnotationsMap
+		recommendName        string
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go]: Removed

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+			c := NewResourceUpdatesCalculator(&frp)
+			patches, err := c.CalculatePatches(tc.pod, test.VerticalPodAutoscaler().WithContainer("test").WithName("name").Get())
+			if tc.expectError == nil {
+				assert.NoError(t, err)
+			} else {
+				if assert.Error(t, err) {
+					assert.Equal(t, tc.expectError.Error(), err.Error())
+				}
+			}
+			if assert.Equal(t, len(tc.expectPatches), len(patches), fmt.Sprintf("got %+v, want %+v", patches, tc.expectPatches)) {
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go]: Done

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+				for i, gotPatch := range patches {
+					if !EqPatch(gotPatch, tc.expectPatches[i]) {
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go]: Same as above, we would miss index info

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go [hunk 8]
```diff
@@ -53,289 +39,137 @@ func (fpp *fakePodPreProcessor) Process(pod apiv1.Pod) (apiv1.Pod, error) {
...
-	}
-}
-
-func addLimitsPatch(idx int) resource_admission.PatchRecord {
-	return resource_admission.PatchRecord{
-		"add",
-		fmt.Sprintf("/spec/containers/%d/resources/limits", idx),
-		apiv1.ResourceList{},
-	}
+func (m fakeVpaMatcher) GetMatchingVPA(_ *apiv1.Pod) *vpa_types.VerticalPodAutoscaler {
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: Done

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+			}},
+		},
+	}
+	recommendAnnotations := vpa_api_util.ContainerToAnnotationsMap{}
+	frp := fakeRecommendationProvider{recommendResources, recommendAnnotations, nil}
+	c := NewResourceUpdatesCalculator(&frp)
+	patches, err := c.CalculatePatches(pod, test.VerticalPodAutoscaler().WithName("name").WithContainer("test").Get())
+	assert.NoError(t, err)
+	// Order of updates for cpu and unobtanium depends on order of iterating a map, both possible results are valid.
+	if assert.Equal(t, len(patches), 3) {
```
[the remark this answers is not in the document]
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/resource_updates_test.go]: Done

**Comment by bskiba** [comment]
Addressed comments. 
What is wrong with github checks? Where is the ci job check?

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (krzysied) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: I like to have coherent coding style.
Plus, it had separate-brackets style previously, so why we want to change (and why only in this place)? It causes to many unneeded questions :)

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (krzysied) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/handler_test.go]: Ack.
I will think about and maybe I will find a solution for more generic Asserts. In a separate PR.

**Pull request merged**